docs/source/store-and-retrieve/object-store.md

```diff
@@ -63,6 +77,10 @@ class ObjectStore(ABC):
     @abstractmethod
     async def delete_object(self, key: str) -> None:
         ...
+
+    @abstractmethod
+    async def list_objects(self, prefix: str | None = None) -> list["ObjectStoreListItem"]:
```

Comment:
I don't see how this can safely be generalized across all object stores without introducing an attack vector.